org.omg.kerml.xtext/src/org/omg/kerml/xtext/library/PrecalculatedLibraryIndexProvider.java

```diff
@@ -78,13 +78,13 @@ public LibraryIndex getIndexFor(Resource resource) {
 			
 			if (indexFile == null || !indexFile.exists()) return LibraryIndex.EMPTY_INDEX;
 			
-			try (FileReader fileReader = new FileReader(indexFile)){
-				
-				index = LibraryIndex.fromJson(fileReader);
-				
+			try {
+				FileInputStream fis = new FileInputStream(indexFile);
+				InputStreamReader isr = new InputStreamReader(fis, StandardCharsets.UTF_8);				
+				index = LibraryIndex.fromJson(isr);
```

Comment:
Previously, the `fileReader` declaration was in a try-with-resources. The `FileInputStream fis` and `InputStreamReader isr` declarations are now within the try body, but they don't ever seem to be closed. Shouldn't the be declared in a try-with-resources, like `fileReader` was? I.e.,
```
try (FileInputStream fis = new FileInputStream(indexFile);
       InputStreamReader isr = new InputStreamReader(fis, StandardCharsets.UTF_8)) {				
    index = LibraryIndex.fromJson(isr);
} catch ...
}
```

Reply:
Opps.  You are right.  I learned try-with-resource in the document:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html